src/test/java/sqlline/SqlLineArgsTest.java

```diff
@@ -2225,6 +2228,91 @@ public void testStartupArgsWithoutUrl() {
     }
   }
 
+  /**
+   * Test to Set/Reset confirm.
+   */
+  @Test
+  public void testConfirmFlag() {
+    DispatchCallback callback = new DispatchCallback();
+    SqlLine line = new SqlLine();
+    assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+    line.runCommands(
+            Collections.singletonList("!set confirm true"), callback);
+    assertEquals(Boolean.TRUE, line.getOpts().getConfirm());
+    line.runCommands(
+            Collections.singletonList("!set confirm false"), callback);
+    assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+  }
+
+  /**
+   * Test effects of confirm flag.
+   */
+  @Test
+  public void testConfirmFlagEffects() {
+    new MockUp<sqlline.Commands>() {
+      @Mock
+      private int getUserAnswer(
+              String question, int... allowedAnswers) throws IOException {
+        return 'y';
+      }
+    };
+    SqlLine line = new SqlLine();
+    assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+    assertEquals("default", line.getOpts().getConfirmPattern());
+    String script = "CREATE TABLE dummy_test(pk int);\n"
+            + "INSERT INTO dummy_test(pk) VALUES(1);\n"
+            + "INSERT INTO dummy_test(pk) VALUES(2);\n"
+            + "INSERT INTO dummy_test(pk) VALUES(3);\n"
+            + "DELETE FROM dummy_test;\n"
+            + "DROP TABLE dummy_test;\n";
+    checkScriptFile(script, true,
+            equalTo(SqlLine.Status.OK),
+            containsString(" "));
+    script = "!set confirm true\n" + script;
+    checkScriptFile(script, true,
+            equalTo(SqlLine.Status.OK),
+            containsString(" "));
+
+  }
+
+  /**
+   * Test effects of confirm flag.
+   */
+  @Test
+  public void testConfirmPattern() {
+    DispatchCallback callback = new DispatchCallback();
+    SqlLine line = new SqlLine();
+    assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+    line.runCommands(
+            Collections.singletonList("!set confirmPattern "
+                    + "\"^(?i:(TRUNCATE|ALTER))\""), callback);
+    assertEquals("^(?i:(TRUNCATE|ALTER))", line.getOpts()
+            .getConfirmPattern());
+    line.runCommands(
+            Collections.singletonList("!set confirmPattern default"), callback);
+    assertEquals("default", line.getOpts().getConfirmPattern());
+
+  }
+
+  /**
+   * Test for illegal regex in confirmPattern
+   */
+  @Test
+  public void testExceptionConfirmPattern() {
+    DispatchCallback callback = new DispatchCallback();
+    SqlLine line = new SqlLine();
+    assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage(containsString("invalid regex"));
+    line.runCommands(
+            Collections.singletonList("!set confirmPattern "
+                    + "\"^(?i*:(TRUNCATE|ALTER))\""), callback);
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage(containsString("null"));
```

Comment:
Do not understand why null is here
What do you think about `default`?

Reply:
If !set confirmPattern is empty, it will treat it as NA or null but not default. This is as per the design that I thought of.

Reply:
But what is the reason for that from user's point of view? What benefit can user get?
Now it looks like the initial value could be `null` but it is forbidden to set `null` and `!set confirmPattern default` leads to a different value (not the initial)